### PR TITLE
feat: switch to sass modules

### PR DESCRIPTION
**Pull request by crisbeto**
Switches all of the .scss files to the Sass module system and to use the new Material theming API.

### CODE REVIEW

File: src/app/pages/component-viewer/component-viewer.scss [hunk 1]
```diff
@@ -1,20 +1,20 @@
-@import 'constants';
+@use '../../../styles/constants';
```
Comment (Splaktar) [inline src/app/pages/component-viewer/component-viewer.scss:1]: Is `@use` not compatible with? Or I guess, does `sass-loader` not support `stylePreprocessorOptions` with `@use`?

https://github.com/angular/material.angular.io/blob/8583f02ea14213d02a354a09cf82cbe0ab1d311a/angular.json#L73-L75

**Review by Splaktar — APPROVED** [review]
Love to see the return of normal `ng serve` and `ng build` times!

Hoping that `stylePreprocessorOptions` can still be supported.

File: src/app/pages/component-viewer/component-viewer.scss (hunk 1 above)
Comment (crisbeto) [reply src/app/pages/component-viewer/component-viewer.scss:1]: I'm guessing it's a `sass-loader` issue, but I think that I prefer the relative path anyway. I found the shorthand confusing at first.

Comment (Splaktar) [reply src/app/pages/component-viewer/component-viewer.scss:1]: I just tested it out and it does look like `sass-loader` and WebStorm are both happy with using `@use` with `stylePreprocessorOptions`. I'd prefer to continue using these shorter paths if they don't cause some kind of build error.

**Pull request merged**